CMakeLists.txt

```diff
@@ -55,59 +55,60 @@ target_link_libraries (nanocat nanomsg)
 
 set (all_tests "")
 
-macro (add_libnanomsg_test NAME)
+macro (add_libnanomsg_test NAME TIMEOUT)
     list (APPEND all_tests ${NAME})
     add_executable (${NAME} tests/${NAME}.c)
     target_link_libraries (${NAME} nanomsg)
     add_test (NAME ${NAME} COMMAND ${NAME})
-    set_tests_properties (${NAME} PROPERTIES  TIMEOUT 30)
+    set_tests_properties (${NAME} PROPERTIES TIMEOUT ${TIMEOUT})
 endmacro (add_libnanomsg_test)
 
 #  Transport tests.
-add_libnanomsg_test (inproc)
-add_libnanomsg_test (inproc_shutdown)
-add_libnanomsg_test (ipc)
-add_libnanomsg_test (ipc_shutdown)
-add_libnanomsg_test (ipc_stress)
-add_libnanomsg_test (tcp)
-add_libnanomsg_test (tcp_shutdown)
-add_libnanomsg_test (ws)
-add_libnanomsg_test (tcpmux)
+add_libnanomsg_test (inproc 5)
+add_libnanomsg_test (inproc_shutdown 5)
+add_libnanomsg_test (ipc 5)
+add_libnanomsg_test (ipc_shutdown 30)
+add_libnanomsg_test (ipc_stress 5)
+add_libnanomsg_test (tcp 5)
+add_libnanomsg_test (tcp_shutdown 120)
```

Comment:
> Do we seriously think this is that load dependent? I'm doubtful.

For clarity, I'm not correlating with the load of the *machine*, but rather the load of the AppVeyor *cluster*, which might conditionally provision more/less resources (this is conjecture). It's hard to draw any conclusions though -- even just 25days ago, the AppVeyor documentation went from 2cores to 1core: https://github.com/appveyor/website/commit/349b45d307045ecfed57382c9606895972f206a8

I can repro on a Windows development VM when configuring that VM to launch with just 1 core. I have not been able to repro on a dev machine with 2+ cores.

Do you have a dev environment in a VM you could configure to just 1 core to see if it's just a Windows problem? (I have OS X on metal, but nothing handy as a VM that could be throttled)